E0194 needs to be updated to new format

From: src/test/compile-fail/E0194.rs

Error E0194 needs a span_label, updating it from: 

```
error[E0194]: type parameter `T` shadows another type parameter of the same name
  --> src/test/compile-fail/E0194.rs:13:5
   |
13 |     fn do_something_else<T: Clone>(&self, bar: T); //~ ERROR E0194
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0194]: type parameter `T` shadows another type parameter of the same name
  --> src/test/compile-fail/E0194.rs:13:5
   |
13 |     fn do_something_else<T: Clone>(&self, bar: T); //~ ERROR E0194
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `T` shadows another type parameter
```

Bonus: the span can be shortened and we can reference the other `T`

```
error[E0194]: type parameter `T` shadows another type parameter of the same name
  --> src/test/compile-fail/E0194.rs:13:5
   |
11 | trait Foo<T> {
   |           - first `T` declared here
12 |     fn do_something(&self) -> T;
13 |     fn do_something_else<T: Clone>(&self, bar: T); //~ ERROR E0194
   |                          ^ shadows another type parameter
```

--- a/src/librustc_typeck/check/wfcheck.rs
+++ b/src/librustc_typeck/check/wfcheck.rs
@@ -658,7 +658,9 @@ fn error_392<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>, span: Span, param_name: ast::N
 }
 
 fn error_194(tcx: TyCtxt, span: Span, name: ast::Name) {
-    span_err!(tcx.sess, span, E0194,
+    struct_span_err!(tcx.sess, span, E0194,
               "type parameter `{}` shadows another type parameter of the same name",
-              name);
+              name)
+        .span_label(span, &format!("`{}` shadows another type parameter", name))
+        .emit();
 }

--- a/src/test/compile-fail/E0194.rs
+++ b/src/test/compile-fail/E0194.rs
@@ -10,7 +10,9 @@
 
 trait Foo<T> {
     fn do_something(&self) -> T;
-    fn do_something_else<T: Clone>(&self, bar: T); //~ ERROR E0194
+    fn do_something_else<T: Clone>(&self, bar: T);
+    //~^ ERROR E0194
+    //~| NOTE `T` shadows another type parameter
 }
 
 fn main() {

